### ISSUE TITLE
`vue/no-undef-components` rule not working for specific component named 'Image'

### ISSUE THREAD

**Issue by shiruten** [vuejs/eslint-plugin-vue#2654]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:*8.14.0* 
- **eslint-plugin-vue version:*8.7.1* 
- **Vue version:*3.2.25* 
- **Node version:*16.14.2* 
- **Operating System:*macOS Sonoma* 

**Please show your full configuration:**

```yml
env:
  browser: true
  es2022: true
  'vue/setup-compiler-macros': true
extends:
  - 'plugin:vue/vue3-recommended'
  - 'eslint:recommended'
  - '@vue/typescript/recommended'
  - 'prettier'
  - 'plugin:storybook/recommended'
parserOptions:
  ecmaVersion: 13
plugins:
  - vue
  - '@typescript-eslint'
rules: { 'vue/multi-word-component-names': 'off', 'vue/no-undef-components': 'error' }
globals: { '__dirname': true }
```

**What did you do?**	

```vue
<script setup>
</script>

<template>
  <Image :path-pc="'/path/to/image.png'" alt-text="Sample" />
</template>
```

**What did you expect to happen?**
ESLint should report an error for the undefined 'Image' component.

**What actually happened?**

ESLint does not report an error for the 'Image' component, even though it is not imported or globally registered. Other undefined components correctly trigger an error.

**Repository to reproduce this issue**

Reproduction repository is currently unavailable, but the issue can be replicated by following the provided steps.

**Additional Context**

- The issue occurs consistently.
- No evidence of special handling for 'Image' in the eslint-plugin-vue source code.
    - https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/no-undef-components.js
- Other undefined components trigger the expected error.(ex. Title, Button)

**Potential Investigation Points**
- Check for any special case handling of 'Image' in eslint-plugin-vue.
- Investigate if the naming convention leads to a bypass of the rule.

**Comment by FloEdelmann** [comment]
`image` is a valid [SVG element](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image):

https://github.com/vuejs/eslint-plugin-vue/blob/926d6e8e5f4a9d953cfb334bb1c1c90d731b86b8/lib/utils/svg-elements.json#L43

That's why it is ignored in the `vue/no-undef-components` rule:

https://github.com/vuejs/eslint-plugin-vue/blob/926d6e8e5f4a9d953cfb334bb1c1c90d731b86b8/lib/rules/no-undef-components.js#L151

https://github.com/vuejs/eslint-plugin-vue/blob/926d6e8e5f4a9d953cfb334bb1c1c90d731b86b8/lib/utils/index.js#L1017-L1019

**Comment by shiruten** [comment]
Here's a thank you response:

Thank you for your quick clarification! I didn't realize that `image` was a valid SVG element. This explains why the ESLint rule wasn't triggering an error.

I'll change the component name to follow Vue's multi-word component naming convention (like `ImageComponent` or `CustomImage`) to avoid confusion with the SVG element.

This was very helpful, thanks again!

**Comment by shiruten** [comment]
Here's a thank you response:

Thank you for your quick clarification! I didn't realize that `image` was a valid SVG element. This explains why the ESLint rule wasn't triggering an error.

I'll change the component name to follow Vue's multi-word component naming convention (like `ImageComponent` or `CustomImage`) to avoid confusion with the SVG element.

This was very helpful, thanks again!
